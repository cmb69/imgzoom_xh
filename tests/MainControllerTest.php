<?php

namespace Imgzoom;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Plib\FakeRequest;
use Plib\View;

class MainControllerTest extends TestCase
{
    public function testShowsImageViewer()
    {
        $sut = new MainController(
            "./",
            "../../userfiles/images/",
            new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["imgzoom"])
        );
        $request = new FakeRequest(["url" => "http://example.com/?&imgzoom_image=foo.jpg"]);
        $response = $sut->defaultAction($request);
        $this->assertSame("text/html; charset=UTF-8", $response->contentType());
        Approvals::verifyHtml($response->output());
    }
}