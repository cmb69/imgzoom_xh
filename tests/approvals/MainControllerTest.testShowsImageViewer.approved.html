<!DOCTYPE html>
<html class="imgzoom_view">
    <head>
        <title>foo.jpg</title>
        <link rel="stylesheet" type="text/css" href="./css/stylesheet.css">
    </head>
    <body>
        <img src="../../userfiles/images/foo.jpg" alt="foo.jpg">
        <script type="text/javascript" src="./imgzoom.js"></script>
    </body>
</html>
